### PR TITLE
round TimestampedAnnotationList fields to nearest representable EDF value

### PR DESCRIPTION
**Pull request by jrevels**

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -94,6 +100,15 @@ struct TimestampedAnnotationList
...
     annotations::Vector{String}
+    function TimestampedAnnotationList(onset_in_seconds, duration_in_seconds, annotations)
+        onset_in_seconds = _nearest_representable_edf_time_value(onset_in_seconds)
+        duration_in_seconds = _nearest_representable_edf_time_value(duration_in_seconds)
+        return new(onset_in_seconds, duration_in_seconds, annotations)
+    end
+end
+
+function _nearest_representable_edf_time_value(x)
+    return round(x, digits=(8 - (ndigits(floor(Int, x)) + signbit(x) + isinteger(x))))
```
Comment (ararslan) [inline src/types.jl:111]:
```suggestion
    return round(x; digits=(8 - (ndigits(floor(Int, x)) + signbit(x) + isinteger(x))))
```
Not every day you get to do a style suggestion for the person who wrote the style guide :stuck_out_tongue:

Comment (ararslan) [reply src/types.jl:111]: CI is failing because `floor` is getting a `nothing` somewhere :thinking:

Comment (ararslan) [reply src/types.jl:111]: Ah, I see, `duration_in_seconds` is allowed to be `nothing`. So you could add a method for `::Nothing` that just returns `nothing`.

**Review by ararslan — APPROVED** [review]
With every passing moment, my hatred for the EDF format grows deeper. This bug is but kindling for the flames of my hate.

File: Project.toml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 name = "EDF"
 uuid = "ccffbfc1-f56e-50fb-a33b-53d1781b2825"
 authors = ["Beacon Biosignals, Inc."]
-version = "0.5.1"
+version = "0.6.0"
```
Comment (ararslan) [inline Project.toml:4]: I'd argue that this is a bugfix, but I guess it could potentially be breaking in edge cases?

Comment (jrevels) [reply Project.toml:4]: Yeah, the previous behavior let you actually construct/manipulate TimestampedAnnotationLists with "unrepresentable" values even if you couldn't write them, so technically is breaking; figure better safe than sorry

**Pull request merged**